src/Uri.php

```diff
@@ -271,6 +270,17 @@ private static function isNonStandardPort(string $scheme, int $port): bool
         return !isset(self::SCHEMES[$scheme]) || $port !== self::SCHEMES[$scheme];
     }
 
+    private function filterHost($host): string
+    {
+        return \preg_replace_callback(
```

Comment:
this implementation is going to be very slow.
My suggestion instead:
strtr($host, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz');